src/OrchardCore/OrchardCore.Abstractions/Json/Nodes/JNode.cs

```diff
@@ -170,26 +171,25 @@ public static bool TryParse(string json, out JsonNode? jsonNode, JsonNodeOptions
     public static string? GetNormalizedPath(this JsonNode jsonNode) => jsonNode.GetPath().GetNormalizedPath();
 
     /// <summary>
-    /// Selects a <see cref="JsonNode"/> from this <see cref="JsonObject"/> using a JSON path.
+    /// Selects a <see cref="JsonNode"/> from this <see cref="JsonObject"/> using JSONPath.
     /// </summary>
-    public static JsonNode? SelectNode(this JsonNode? jsonNode, string? path)
+    /// <param name="jsonNode">The JSON node which serves as the root of the current search..</param>
+    /// <param name="path">The JSONPath query where <c>$</c> is <paramref name="jsonNode"/>.</param>
+    /// <param name="options">Optional settings to configure the JSONPath parser.</param>
+    /// <remarks>
+    /// This method uses JsonPath.Net to evaluate the <paramref name="path"/>. For more information on JSONPath, see the
+    /// specification at https://www.rfc-editor.org/rfc/rfc9535.html or the JsonPath.Net documentation at
+    /// https://docs.json-everything.net/path/basics/.
+    /// </remarks>
+    public static JsonNode? SelectNode(this JsonNode? jsonNode, string? path, PathParsingOptions? options = null)
     {
-        if (jsonNode is null || path is null)
-        {
-            return null;
-        }
-
-        if (jsonNode is JsonObject jsonObject)
-        {
-            return jsonObject.SelectNode(path);
-        }
-
-        if (jsonNode is JsonArray jsonArray)
-        {
-            return jsonArray.SelectNode(path);
-        }
+        path = path?.Trim();
+        if (string.IsNullOrEmpty(path)) return jsonNode;
+        if (path[0] != '$') path = "$." + path;
```

Comment:
is not using `$.` not supported in JSONPath.Net?
Ideally we would not patch what the caller specifies, but if we are using something that is currently incompatible, and JSONPath.NEt doesn't support it then fine. Add a comment in that case.

Reply:
It's not supported. (afaik paths without the leading `$` is non-spec in JSONPath) Without that condition `JsonPath.TryParse` returns false, and `JsonPath.Parse(path)` throws:
```
Json.Path.PathParseException: Path must start with '$'
   at Json.Path.PathParser.Parse(ReadOnlySpan`1 source, Int32& index, PathParsingOptions options, Boolean requireGlobal)
   at Json.Path.JsonPath.Parse(String source, PathParsingOptions options)
```
Based on the exception message, this patch should be universally safe.

I believe in Newtonsoft does this automatically under the hood. You can see from this example, that the path `property` behaves as `$.property` and not as `$..property`.

![image](https://github.com/OrchardCMS/OrchardCore/assets/4007293/d8207419-2214-47b5-8e23-b1cffad5887f)

I agree that it's not ideal, but without this patch existing functionality in OC would break.